flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -541,10 +549,12 @@ public ExecutionConfig setExecutionRetryDelay(long executionRetryDelay) {
      * Sets the execution mode to execute the program. The execution mode defines whether data
      * exchanges are performed in a batch or on a pipelined manner.
      *
-     * <p>The default execution mode is {@link ExecutionMode#PIPELINED}.
-     *
+     * @deprecated This configuration is used in the deprecated DataSet API. The {@link
+     *     ExecutionConfig#setJobType(String)} sets execution runtime mode of job
+     *     <p>The default execution mode is {@link ExecutionMode#PIPELINED}.
      * @param executionMode The execution mode to use.
      */
+    @Deprecated
```

Comment:
Add the `@Deprecated` for EXECUTION_MODE  should be a separate commit.

You can take a look the commit part of `Code Style and Quality Guide`[1].

[1] https://flink.apache.org/contributing/code-style-and-quality-pull-requests.html#separate-refactoring-cleanup-and-independent-changes